src/visual_tool.cpp

```diff
@@ -273,14 +268,15 @@ void VisualTool<FeatureType>::OnMouseEvent(wxMouseEvent &event) {
 
 template<class FeatureType>
 void VisualTool<FeatureType>::DrawAllFeatures() {
-	gl.SetLineColour(colour[0], 1.0f, 1);
+	wxColour grid_color = to_wx(OPT_GET("Colour/Visual Tools/Lines Secondary")->GetColor());
+	gl.SetLineColour(grid_color, 1.0f, 1);
 	for (auto& feature : features) {
-		int fill = 1;
+		wxColour fill = to_wx(OPT_GET("Colour/Visual Tools/Highlight Primary")->GetColor());
```

Comment:
These should be pulled out of the loop. Options lookups aren't particularly optimized and you can have a very large number of features for complex clips.

Reply:
Maybe even just put all the `OPT_GET`s in the constructors and store pointers to the `agi::OptionValue` as member variables (for all of the visual tools).